File: Python/InsertionSort.py
# Insertion Sort in Python, Time Complexity: O(n^2)

def displayElements(arr, n):
    for el in arr:
        print(el, " ", end = "")
    print()
    
def insertionSort(arr): 
    for i in range(1, len(arr)): 
        key = arr[i] 
  
        j = i - 1
        while j >= 0 and key < arr[j] : 
                arr[j + 1] = arr[j] 
                j -= 1
        arr[j + 1] = key


if __name__ == "__main__":
    print("Enter number of array elements: ")
    n = int(input())
    print("Enter array elements: ")
    arr = list(map(int, input().split()))

    # Array before Insertion Sort
    print("Array before Insertion Sort: ")
    displayElements(arr, n)
    
    insertionSort(arr)

    # Array after Insertion Sort
    print("Array after Insertion Sort: ")
    displayElements(arr, n)

File: Java/InsertionSort.java
// Insertion Sort in Java, Time Complexity: O(n^2)

import java.io.*;
import java.util.*;

public class InsertionSort { 
   
    static void displayElements(int arr[]) 
    { 
        int n = arr.length; 
        for (int i = 0; i < n; ++i) 
            System.out.print(arr[i] + " ");
        System.out.println(); 
    }
    
    void insertion_sort(int arr[]) 
    { 
        int n = arr.length; 
        for (int i = 1; i < n; i++) { 
            int key = arr[i]; 
            int j = i - 1; 
  
            while (j >= 0 && arr[j] > key) { 
                arr[j + 1] = arr[j]; 
                j -= 1; 
            } 
            arr[j + 1] = key; 
        } 
    } 
  
    public static void main(String args[]) 
    { 
        Scanner sc = new Scanner(System.in);
        
        int n;
        System.out.println("Enter the number of array elements: ");
        n = sc.nextInt();
        
        int arr[] = new int[n];
        System.out.println("Enter the array elements: ");
        for(int i = 0; i < n; i++)
            arr[i] = sc.nextInt();
        
        System.out.println("Array elements before Insertion Sort: ");
        displayElements(arr); 
        
        InsertionSort obj = new InsertionSort(); 
        obj.insertion_sort(arr); 
  
        System.out.println("Array elements after Insertion Sort: ");
        displayElements(arr); 
    } 
}